### PR TITLE
Milestone 6: Feature MusicAlbum and Genre

### PR DESCRIPTION
**Pull request by dqarias**
In this milestone we:
- Create MusicAlbum class in a separate .rb file.
- Create Genre class with an association to the Item class (in a separate .rb file).
- All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.
- All Genre class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Genre class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the MusicAlbum class
        - should override the method from the parent class
        - should return true if parent's method returns true AND if on_spotify equals true
        - otherwise, it should return false

### CODE REVIEW

File: lib/music_album.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    @on_spotify = on_spotify
+  end
+
+  private
+
+  def can_be_archieved?
+    return true if super && @on_spotify == true
+
+    false
+  end
```
Comment (darikmohammed) [inline lib/music_album.rb:17]: [Optional] you can write the function with `super && @on_spotify`  it will do the same thing as your implementation.

**Review by darikmohammed — APPROVED** [review]
Hi @dqarias , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code
:heavy_check_mark: No linter error

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: lib/music_album.rb (hunk 1 above)
Comment (dqarias) [reply lib/music_album.rb:17]: @darikmohammed thanks for your feedback

**Review by Olamarx — APPROVED** [review]
Hi @dqarias, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Organized code
:heavy_check_mark: No linter error

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**